File: TrabajoI.C/Java/claseabstracta/Producto.java
package claseabstracta;

public class Producto extends Elemento {
    String name;
    double precio;
    private String nombre;

    Producto(String nombre, double precio) {   // Constructor con parámetros
        this.nombre = nombre;
        this.precio = precio;
    }

    @Override
    void mostrarInfo() {
        System.out.println("Producto: " + nombre + ", Precio: $" + precio);
    }
}

File: TrabajoI.C/Go/claseAbstracta/Producto.go
type Producto struct {
	entidadBase
	Nombre string
	Precio float64
}


// Constructor copia
func NewProductoCopy(p *Producto) *Producto {
	return &Producto{entidadBase: newEntidadBase(p.ID), Nombre: p.Nombre, Precio: p.Precio}
}

func (p *Producto) Info() string {
	return "Producto: " + p.Nombre
}
